orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -125,15 +125,15 @@ public void verifyTrafficRemoval(String serverGroupName, Moniker serverGroupMoni
     List<TargetServerGroup> targetServerGroups = ((List<Map<String, Object>>) cluster.get().get("serverGroups"))
       .stream()
       .map(TargetServerGroup::new)
-      .filter(tsg -> location.equals(tsg.getLocation()))
+      .filter(tsg -> location.equals(tsg.getLocation(location.getType())))
```

Comment:
@ewiseblatt this is the fix attempt we were discussing earlier

Reply:
ok. thanks for pointing that out. I thought it was a different one.